plugins/PrivacyManager/Model/DataSubjects.php

```diff
@@ -0,0 +1,395 @@
+
+        foreach ($logTables as $logTable) {
+            $logTableName = $logTable->getName();
+            if ($logTableName === 'log_action') {
+                continue; // will be deleted via cron job automatically if the action is no longer in use
+            }
+
+            $from = array($logTableName);
+            $tableToSelect = $this->findNeededTables($logTable, $from);
+
+            if (!$tableToSelect) {
+                throw new \Exception('Cannot join table ' . $logTable->getName());
+            }
+
+            list($where, $bind) = $this->visitsToWhereAndBind($tableToSelect, $visits);
+
+            $sql = "DELETE $logTableName FROM " . $this->makeFromStatement($from) . " WHERE $where";
+
+            $result = Db::query($sql, $bind)->rowCount();
+
+            $results[$logTableName] = $result;
+        }
+
+        krsort($results); // make sure test results are always in same order
+
+        return $results;
+    }
+
+    /**
+     * @param LogTable[] $logTables
+     * @return LogTable[]
+     */
+    private function sortLogTablesToEnsureDataErasureFromAllTablesIsPossible($logTables)
+    {
+        // we need to make sure to delete eg first entries from a table log_form_field before log_form...
+        // otherwise when deleting log_form entries first, the log_form_field entries that belong to a requested visit
+        // cannot be deleted anymore as the links would be all gone. We need to make sure to delete from log_visit last
+        // and log_link_visit_action second to last.
+        usort($logTables, function ($a, $b) {
+            /** @var LogTable $a */
+            /** @var LogTable $b */
+            $aName = $a->getName();
+            $bName = $b->getName();
+            if ($bName === 'log_visit') {
+                return -1;
+            } else if ($aName === 'log_visit') {
+                return 1;
+            } else if ($bName === 'log_link_visit_action') {
+                return -1;
+            } else if ($aName === 'log_link_visit_action') {
+                return 1;
+            }
+
+            $aWays = $a->getWaysToJoinTable();
+            foreach ($aWays as $table => $column) {
+                if ($table === $bName) {
+                    return -1;
+                }
+            }
+
+            $bWays = $b->getWaysToJoinTable();
+            foreach ($bWays as $table => $column) {
+                if ($table === $aName) {
+                    return 1;
+                }
+            }
+
+            if ($bWays && !$aWays) {
+                return 1;
+            }
+
+            if (!$bWays && $aWays) {
+                return -1;
+            }
+
+            return 0;
+        });
+        return $logTables;
+    }
+
+    public function exportDataSubjects($visits)
+    {
+        if (empty($visits)) {
+            return array();
+        }
+
+        $logTables = $this->logTablesProvider->getAllLogTables();
```

Comment:
unfortunately we cannot use logQueryBuilder here because of 3rd party plugins and needing to execute queries differently.